### DISCUSSION TITLE
The difference between two validation metrics in file log❓ [QUESTION]

### DISCUSSION THREAD

**Discussion by fandengdong** [mir-group/nequip#337] [Q&A] [answered]
Hi, Nequip is a very nice work! I'm trying to reproduce the result using the official codes currently. I found there are two validation metrics in the log file. The difference between them is quit large (see below):

![image](https://github.com/mir-group/nequip/assets/38948614/549feb5d-e7c9-4bcd-9cf5-de577fd2b0c8)

What is the difference between the parts enclosed by the red box in the picture?

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @whu-dft ,

Thanks for your interest in our work!

The upper metrics are averages across a single minibatch (see the "batch" column).  The later ones are averages over the _entire_ epoch (note the column is now epoch and not batch). Note also that `nequip` correctly does the whole epoch averages as over all samples in the epoch, and **not** as a mean of means, which is incorrect.

There is only one log line during the epoch because you have fewer batches than the log frequency, so you only get one line.
